### ISSUE TITLE
Please integrate SHA3 version 1.0.2 with Raspberry support

### ISSUE THREAD

**Issue by StephanHarrer** [rubygems/rubygems#4184]
### Describe the problem as clearly as you can

Hello it seems Bundler is still using sha3 1.0.1. 1.0.1 is without support for Raspberry Pi, this is added with version 1.0.2.
Please integrate this new version (https://rubygems.org/gems/sha3/versions/1.0.2).

**Comment by deivid-rodriguez** [comment]
Hi @StephanHarrer, could you point exactly where we're using this outdated code, and how you found about this issue? Thanks.

**Comment by StephanHarrer** [comment]
Hello Deivid,

I do the following on an raspberry. As far as I understand I run into an error because sha3 -v '1.0.1' is not installed. However, there are problems with version 1.0.1 on a Raspberry. Therefore Johanns Gregorian has fortunately released a version 1.0.2 (https://rubygems.org/gems/sha3/versions/1.0.1) which runs on the Raspberry.

However, the error does not disappear when I install the version sha 1.0.2. 


_stephan@rasp4:~/dnscat2/server$ sudo gem install bundler
stephan@rasp4:~/dnscat2/server$ sudo bundle install

_Don't run Bundler as root. Bundler can ask for sudo if it is needed, and installing your bundle as root will break this application for all non-root users on this machine.
Fetching gem metadata from https://rubygems.org/.......
Using bundler 2.2.3
Using trollop 2.1.2
Using salsa20 0.1.1
Using ecdsa 1.2.0
Fetching sha3 1.0.1
Installing sha3 1.0.1 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /var/lib/gems/2.7.0/gems/sha3-1.0.1/ext/sha3
/usr/bin/ruby2.7 -I /usr/lib/ruby/2.7.0 -r ./siteconf20201223-190102-bsp3pb.rb extconf.rb
checking for sha3.h... yes
checking for digest.h... yes
creating Makefile

current directory: /var/lib/gems/2.7.0/gems/sha3-1.0.1/ext/sha3
make "DESTDIR=" clean

current directory: /var/lib/gems/2.7.0/gems/sha3-1.0.1/ext/sha3
make "DESTDIR="
compiling KeccakF-1600-reference.c
cc1: error: unknown value ‘nocona’ for ‘-march’
cc1: note: valid arguments are: armv8-a armv8.1-a armv8.2-a armv8.3-a armv8.4-a armv8.5-a armv8.6-a native
make: *** [Makefile:245: KeccakF-1600-reference.o] Error 1

make failed, exit code 2

Gem files will remain installed in /var/lib/gems/2.7.0/gems/sha3-1.0.1 for inspection.
Results logged to /var/lib/gems/2.7.0/extensions/aarch64-linux/2.7.0/sha3-1.0.1/gem_make.out

An error occurred while installing sha3 (1.0.1), and Bundler cannot continue.
Make sure that `gem install sha3 -v '1.0.1' --source 'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  sha3_
stephan@rasp4:~/dnscat2/server$_

Regards
Stephan

**Comment by deivid-rodriguez** [comment]
If you delete your `Gemfile.lock` file and try again, it should pickup the latest version of the gem.

**Comment by StephanHarrer** [comment]
Hello Deivid,

I can't see any difference.

**Comment by deivid-rodriguez** [comment]
Can you post the `Gemfile`? Can you also run `bundle lock` and post the generated `Gemfile.lock` file?

**Comment by StephanHarrer** [comment]
Gemfile:
# Gemfile
# By Ron Bowes
#
# See LICENSE.md

source 'https://rubygems.org'

gem 'trollop' # Commandline parsing
gem 'salsa20' # Encrypted connections
gem 'sha3'    # Message signing + key derivation
gem 'ecdsa'   # Used for ECDH key exchange

#############################################
Gemfile.lock
GEM
  remote: https://rubygems.org/
  specs:
    ecdsa (1.2.0)
    salsa20 (0.1.1)
    sha3 (1.0.1)
    trollop (2.1.2)

PLATFORMS
  ruby

DEPENDENCIES
  ecdsa
  salsa20
  sha3
  trollop
(END)

**Comment by deivid-rodriguez** [comment]
@StephanHarrer I can't reproduce your issue. This is the lockfile I get:

```
GEM
  remote: https://rubygems.org/
  specs:
    ecdsa (1.2.0)
    salsa20 (0.1.3)
    sha3 (1.0.2)
    trollop (2.9.10)

PLATFORMS
  x86_64-linux

DEPENDENCIES
  ecdsa
  salsa20
  sha3
  trollop

BUNDLED WITH
   2.2.3
```

Can you post your environment details, i.e., the result of `gem env` and `bundle env`?

Please use markdown to format your messages nicely. Here's a guide: https://guides.github.com/features/mastering-markdown/.

**Comment by StephanHarrer** [comment]
stephan@rasp4:~/dnscat2/server$ gem env
RubyGems Environment:
  \- RUBYGEMS VERSION: 3.1.2
  \- RUBY VERSION: 2.7.1 (2020-03-31 patchlevel 83) [aarch64-linux-gnu]
  \- INSTALLATION DIRECTORY: /var/lib/gems/2.7.0
  \- USER INSTALLATION DIRECTORY: /home/stephan/.gem/ruby/2.7.0
  \- RUBY EXECUTABLE: /usr/bin/ruby2.7
  \- GIT EXECUTABLE: /usr/bin/git
  \- EXECUTABLE DIRECTORY: /usr/local/bin
  \- SPEC CACHE DIRECTORY: /home/stephan/.gem/specs
  \- SYSTEM CONFIGURATION DIRECTORY: /etc
  \- RUBYGEMS PLATFORMS:
  &nbsp;&nbsp;&nbsp;&nbsp;\- ruby
  &nbsp;&nbsp;&nbsp;&nbsp;\- aarch64-linux
  \- GEM PATHS:
    &nbsp;&nbsp;&nbsp;&nbsp;\- /var/lib/gems/2.7.0
    &nbsp;&nbsp;&nbsp;&nbsp;\- /home/stephan/.gem/ruby/2.7.0
    &nbsp;&nbsp;&nbsp;&nbsp;\- /usr/local/lib/ruby/gems/2.7.0
    &nbsp;&nbsp;&nbsp;&nbsp;\- /usr/lib/ruby/gems/2.7.0
    &nbsp;&nbsp;&nbsp;&nbsp;\- /usr/lib/aarch64-linux-gnu/ruby/gems/2.7.0
    &nbsp;&nbsp;&nbsp;&nbsp;\- /usr/share/rubygems-integration/2.7.0
    &nbsp;&nbsp;&nbsp;&nbsp;\- /usr/share/rubygems-integration/all
    &nbsp;&nbsp;&nbsp;&nbsp;\- /usr/lib/aarch64-linux-gnu/rubygems-integration/2.7.0
  \- GEM CONFIGURATION:
     &nbsp;&nbsp;&nbsp;&nbsp;\- :update_sources => true
     &nbsp;&nbsp;&nbsp;&nbsp;\- :verbose => true
     &nbsp;&nbsp;&nbsp;&nbsp;\- :backtrace => false
     &nbsp;&nbsp;&nbsp;&nbsp;\- :bulk_threshold => 1000

  \- REMOTE SOURCES:
     &nbsp;&nbsp;&nbsp;&nbsp;\- https://rubygems.org/
  \- SHELL PATH:
      &nbsp;&nbsp;&nbsp;&nbsp;\- /usr/local/sbin
      &nbsp;&nbsp;&nbsp;&nbsp;\- /usr/local/bin
      &nbsp;&nbsp;&nbsp;&nbsp;\- /usr/sbin
      &nbsp;&nbsp;&nbsp;&nbsp;\- /usr/bin
      &nbsp;&nbsp;&nbsp;&nbsp;\- /sbin
      &nbsp;&nbsp;&nbsp;&nbsp;\- /bin
      &nbsp;&nbsp;&nbsp;&nbsp;\- /usr/games
      &nbsp;&nbsp;&nbsp;&nbsp;\- /usr/local/games
      &nbsp;&nbsp;&nbsp;&nbsp;\- /snap/bin

stephan@rasp4:~/dnscat2/server$ bundle env
\#\# Environment

\```
Bundler       2.2.3
  &nbsp;&nbsp;&nbsp;&nbsp;Platforms   ruby, aarch64-linux
Ruby          2.7.1p83 (2020-03-31 revision a0c7c23c9cec0d0ffcba012279cd652d28ad5bf3) [aarch64-linux-gnu]
  &nbsp;&nbsp;&nbsp;&nbsp;Full Path   /usr/bin/ruby2.7
  &nbsp;&nbsp;&nbsp;&nbsp;Config Dir  /etc
RubyGems      3.1.2
  &nbsp;&nbsp;&nbsp;&nbsp;Gem Home    /var/lib/gems/2.7.0
  &nbsp;&nbsp;&nbsp;&nbsp;Gem Path /home/stephan/.gem/ruby/2.7.0:/var/lib/gems/2.7.0:/usr/local/lib/ruby/gems/2.7.0:/usr/lib/ruby/gems/2.7.0:/usr/lib/aarch64-linux-gnu/ruby/gems/2.7.0:/usr/share/rubygems-integration/2.7.0:/usr/share/rubygems-int             egration/all:/usr/lib/aarch64-linux-gnu/rubygems-integration/2.7.0
  &nbsp;&nbsp;&nbsp;&nbsp;User Home   /home/stephan
  &nbsp;&nbsp;&nbsp;&nbsp;User Path   /home/stephan/.gem/ruby/2.7.0
  &nbsp;&nbsp;&nbsp;&nbsp;Bin Dir     /usr/local/bin
Tools
  &nbsp;&nbsp;&nbsp;&nbsp;Git         2.27.0
  &nbsp;&nbsp;&nbsp;&nbsp;RVM         not installed
  &nbsp;&nbsp;&nbsp;&nbsp;rbenv       not installed
  &nbsp;&nbsp;&nbsp;&nbsp;chruby      not installed
\```

\## Bundler Build Metadata

\```
Built At          2020-12-22
Git SHA           29dc3c8398
Released Version  true
\```

\## Gemfile

\### Gemfile

\```ruby
\# Gemfile
\# By Ron Bowes
\#
\# See LICENSE.md

source 'https://rubygems.org'

gem 'trollop' # Commandline parsing
gem 'salsa20' # Encrypted connections
gem 'sha3'    # Message signing + key derivation
gem 'ecdsa'   # Used for ECDH key exchange
\```

\### Gemfile.lock

\```
GEM
  &nbsp;&nbsp;&nbsp;&nbsp;remote: https://rubygems.org/
  &nbsp;&nbsp;&nbsp;&nbsp;specs:
   &nbsp;&nbsp;&nbsp;&nbsp; ecdsa (1.2.0)
   &nbsp;&nbsp;&nbsp;&nbsp; salsa20 (0.1.1)
   &nbsp;&nbsp;&nbsp;&nbsp; sha3 (1.0.1)
   &nbsp;&nbsp;&nbsp;&nbsp; trollop (2.1.2)

PLATFORMS
  &nbsp;&nbsp;&nbsp;&nbsp;ruby

DEPENDENCIES
  &nbsp;&nbsp;&nbsp;&nbsp;ecdsa
  &nbsp;&nbsp;&nbsp;&nbsp;salsa20
  &nbsp;&nbsp;&nbsp;&nbsp;sha3
  &nbsp;&nbsp;&nbsp;&nbsp;trollop
\```
stephan@rasp4:~/dnscat2/server$

**Comment by deivid-rodriguez** [comment]
Please don't ignore the parts of my messages that help me, but attend to the parts that help you, and properly format your command results using markdown. Thanks.

**Comment by StephanHarrer** [comment]
Sorry Deivid, I used markdown and the printout is exactly like in the CLI of my system.

**Comment by deivid-rodriguez** [comment]
Wrap each command with triple backquotes, so that it's nicely formatted.

Anyways, if you're really deleting the `Gemfile.lock` file before running `bundle lock`, and that's what you're getting, and completely lost on what the problem is. If you can build a Dockerfile that demonstrates this behaviour I'm happy to have a look, otherwise there's nothing I can do to help, sorry.

**Comment by StephanHarrer** [comment]
Hello Deivid,

I have found the problem. The installation creates not only the Gemfile but also the Gemfile.lock.
After I have deleted the Gemfile.lock as suggested by you, I have run the installation completely again. Then the Gemfile.lock was created again and in the SSH 1.0.1 is in it.

Now I have edited the file manually and entered the version 1.0.2. With that it works. Thanks for your help and sorry for the inconvenience.

Regards
